torch_xla/experimental/xla_sharding.py

```diff
@@ -492,11 +533,12 @@ class ShardingSpec:
   @xr.requires_pjrt
   def __post_init__(self):
     partition_spec, mesh = self.partition_spec, self.mesh
-    self._tile_assignment = _get_tile_assignment(mesh, partition_spec)
+    tile_assignment, partition_spec = _get_tile_assignment(mesh, partition_spec)
```

Comment:
if this method also returns partition_spec, you should call it something like `_get_tile_assignment_and_update_partition_spec`

Reply:
Good point, maybe I can do a bit more refactoring and not return the partition spec here. It feels a bit clunky to be rewriting the partition spec anyway.

Reply:
You mean in-placed updating the partition spec?

Reply:
Just generally having to modify it. The partition spec has served its purpose in `_get_tile_assignment`, we should be able to get the group assignment without using the partition spec.